### PR TITLE
Lightbox: v2

### PR DESCRIPTION
**Pull request by gblakeman**
https://www.pivotaltracker.com/story/show/152292761

### CODE REVIEW

**Comment by ellodanger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by alanpeabody** [comment]
@gblakeman is this ready for review?

**Comment by gblakeman** [comment]
@alanpeabody Hey, sorry. Short answer, no. Long answer, it’s on my to-do list today to get you all an update on where I’m at with it.

**Comment by gblakeman** [comment]
@alanpeabody This is ready for 👀 – holler if you have any questions. There are certainly things I can simplify/optimize a bit more, particularly in the `LightBoxContainer`, but at this point I’ve deferred some of that to a future round.

File: src/selectors/light_box.js [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+          const assetId = region.getIn(['links', 'assets'])
+          if (assetId) {
+            return combinedPostsAssetIds.push([postId, assetId])
+          }
+          return null
+        })
+      }
+
+      // retrieve `postId` + `assetId` pairing
+      postContent.map((region) => {
```
Comment (alanpeabody) [inline src/selectors/light_box.js]: I think a some of the `.map`s in this function should probably be `each` as we don't actually capture the return, or the function refactored to not use mutation(`push`). But NBD.

Comment (gblakeman) [reply src/selectors/light_box.js]: Cool, I think I need to adjust these for a repost-related bug I just found anywhere.

**Comment by gblakeman** [comment]
I found an issue in related to reposts that have additional content. It shouldn’t be hard to adjust and i’ll try to get to it today. Noting it here so this doesn’t get deployed on Monday if I haven't gotten to it yet.

**Comment by alanpeabody** [comment]
:tada: 🏆 👍 💯 :tada:

**Pull request merged**